### DISCUSSION TITLE
Retracted

### DISCUSSION THREAD

**Discussion by HappyCerberus** [carbon-language/carbon-lang#1606] [Language design]
Retracted.

**Comment by geoffromer** [comment]
> Why does Carbon want to replicate language design failures of other languages?

This kind of rhetorical question shuts down constructive discussion. Please take a look at our [code of conduct](https://github.com/carbon-language/carbon-lang/blob/trunk/CODE_OF_CONDUCT.md), particularly "Be friendly and patient" and "When we disagree, try to understand why".

**Reply by HappyCerberus** [reply]
Hmm:

- I do consider them design failures
- Carbon seems to be copying them
- I do want to know why

But I do retract my question and will no longer participate.

**Comment by OlaFosheimGrostad** [comment]
> considering all language that I know of that started with single inheritance had to backport multiple inheritance of some form

Out of curiosity, which languages are you thinking of?

Simula, that C++ builds on, had single inheritance. Beta, the Simula successor, also had single inheritance. Beta built on a experiences from a modelling language Delta (IIRC). I attended a lecture where Kristen Nygaard layed out the reasoning behind Beta and why they preferred single inheritance, and their reasoning made sense, so I don't think one can jump to any quick conclusions about having a preference for single inheritance. Usage of multiple inheritance is often a sign of poor modelling. That said, Beta also had type variables and virtual types, so you could not only extend the super class, but also extend types within the super class (virtually). Beta also unified classes and functions into one concept, reminiscent of how C++ lambdas are classes. I think Bjarne Stroustrup once said that there was a nice little language hidden within C++, maybe this aspect of Beta was what he was thinking of?
